client/modules/IDE/components/AssetList.jsx

```diff
@@ -1,187 +1,63 @@
-import { useTranslation, withTranslation } from 'react-i18next';
-import MenuItem from '../../../components/Dropdown/MenuItem';
-import TableDropdown from '../../../components/Dropdown/TableDropdown';
-
+import { useTranslation } from 'react-i18next';
+import AssetListRow from './AssetListRow';
 import Loader from '../../App/components/loader';
-import { deleteAssetRequest } from '../actions/assets';
 import * as AssetActions from '../actions/assets';
 
-const AssetMenu = ({ item: asset }) => {
+const AssetList = () => {
   const { t } = useTranslation();
-
   const dispatch = useDispatch();
+  const { username, assetList, loading } = useSelector((state) => ({
+    username: state.user.username,
+    assetList: state.assets.list,
+    loading: state.loading
+  }));
 
-  const handleAssetDelete = () => {
-    const { key, name } = asset;
-    if (window.confirm(t('Common.DeleteConfirmation', { name }))) {
-      dispatch(deleteAssetRequest(key));
-    }
-  };
-
-  return (
-    <TableDropdown aria-label={t('AssetList.ToggleOpenCloseARIA')}>
-      <MenuItem onClick={handleAssetDelete}>{t('AssetList.Delete')}</MenuItem>
-      <MenuItem href={asset.url} target="_blank">
-        {t('AssetList.OpenNewTab')}
-      </MenuItem>
-    </TableDropdown>
-  );
-};
-
-AssetMenu.propTypes = {
-  item: PropTypes.shape({
-    key: PropTypes.string.isRequired,
-    url: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired
-  }).isRequired
-};
-
-const AssetListRowBase = ({ asset, username }) => (
-  <tr className="asset-table__row" key={asset.key}>
-    <th scope="row">
-      <a href={asset.url} target="_blank" rel="noopener noreferrer">
-        {asset.name}
-      </a>
-    </th>
-    <td>{prettyBytes(asset.size)}</td>
-    <td>
-      {asset.sketchId && (
-        <Link to={`/${username}/sketches/${asset.sketchId}`}>
-          {asset.sketchName}
-        </Link>
-      )}
-    </td>
-    <td className="asset-table__dropdown-column">
-      <AssetMenu item={asset} />
-    </td>
-  </tr>
-);
-
-AssetListRowBase.propTypes = {
-  asset: PropTypes.shape({
-    key: PropTypes.string.isRequired,
-    url: PropTypes.string.isRequired,
-    sketchId: PropTypes.string,
-    sketchName: PropTypes.string,
-    name: PropTypes.string.isRequired,
-    size: PropTypes.number.isRequired
-  }).isRequired,
-  username: PropTypes.string.isRequired
-};
-
-function mapStateToPropsAssetListRow(state) {
-  return {
-    username: state.user.username
-  };
-}
-
-function mapDispatchToPropsAssetListRow(dispatch) {
-  return bindActionCreators(AssetActions, dispatch);
-}
-
-const AssetListRow = connect(
-  mapStateToPropsAssetListRow,
-  mapDispatchToPropsAssetListRow
-)(AssetListRowBase);
-
-class AssetList extends React.Component {
-  constructor(props) {
-    super(props);
-    this.props.getAssets();
-  }
-
-  getAssetsTitle() {
-    return this.props.t('AssetList.Title');
-  }
+  useEffect(() => {
+    dispatch(AssetActions.getAssets());
+  }, [dispatch]);
```

Comment:
i feel like since this is from the constructor we probably don't need anything in the dependency array?

Reply:
`dispatch` never changes its value so it doesn't make a difference whether it's in the dependency array or not.  The effect will run one time on mount either way.  Personally I'm in the habit of including it to appease linting rules such as `react-hooks/exhaustive-deps`.  But we don't have that rule enabled on this repo (at least not yet) because we violate it all the time.